qiskit/qpy/__init__.py

```diff
@@ -370,6 +370,30 @@ def open(*args):
 by ``num_circuits`` in the file header). There is no padding between the
 circuits in the data.
 
+.. _qpy_version_14:
+
+Version 14
+----------
+
+Version 14 adds support for additional :class:`~.types.Type` classes, and adds support for
+const-ness to existing types :class:`~.types.Bool` and :class:`~.types.Uint`.
+The ``EXPR_TYPE_BOOL`` and ``EXPR_TYPE_UNIT`` structs are now replaced by ``EXPR_TYPE_BOOL_V14``
+and ``EXPR_TYPE_UINT_V14``, respectively. See the updated expression type table below.
+
+EXPR_TYPE
+~~~~~~~~~
+
+A :class:`~.types.Type` is encoded by a single-byte ASCII ``char`` that encodes the kind of type,
+followed by a payload that varies depending on the type.  The defined codes are:
+
+======================  =========  =================================================================
+Qiskit class            Type code  Payload
+======================  =========  =================================================================
+:class:`~.types.Bool`   ``b``      One `_Bool const`.
+
+:class:`~.types.Uint`   ``u``      One ``uint32_t width``, followed by one ``_Bool const``.
+======================  =========  =================================================================
```

Comment:
Maybe want to be slightly clearer in this that this modifies the meanings of the type keys compared to version 9.  Line 392 needs double backticks around ``_Bool const``.

Reply:
I've changed this around a bit in the later PRs. If it's alright, it might be easier for us to keep this as is and then make sure that the final PR reflects everything in a cohesive and complete manner.

Reply:
I'm fine to defer, but can you open an issue tagged for the milestone about what you mean, just to remind us to do it as part of the documentation in the final PR?

Reply:
https://github.com/Qiskit/qiskit/issues/13927

Reply:
Issue #13927 is obsoleted by 8021e00850116efd16aa6b3688fac46f9d289afe, I think?

Reply:
Well, if not obsoleted, then at least the changes to QPY should now be localised to a later PR I think.